src/v/cloud_storage/remote_segment.cc

```diff
@@ -278,10 +279,18 @@ remote_segment::offset_data_stream(
   kafka::offset start,
   kafka::offset end,
   std::optional<model::timestamp> first_timestamp,
-  ss::io_priority_class io_priority) {
+  ss::io_priority_class io_priority,
+  storage::opt_abort_source_t as) {
```

Comment:
Is there a case, excluding tests, where we call offset_data_stream without an abort source?

Reply:
no, do you thing this new parameter should be non-optional, it would be easy to change tests to take an abort source

Reply:
I think so, it does not provide additional testing functionality and it would make the code simpler to follow.

Reply:
It looks like I'll have to revert this, as the abort source passed to offset data stream (in actual code path) is from log reader config which may not have one, that field is optional.